ownCloud/Theming/ThemeCollection.swift

```diff
@@ -177,68 +186,85 @@ class ThemeCollection : NSObject {
 		self.purchaseColors.disabled.background = self.purchaseColors.disabled.background.greyscale()
+		self.destructiveColors = colors.resolveThemeColorPairCollection("Fill.destructiveColors", ThemeColorPairCollection(fromPair: ThemeColorPair(foreground: UIColor.white, background: UIColor.red)), generic)
 
-		self.tintColor = self.lightBrandColor
+		self.tintColor = colors.resolveColor("tintColor", self.lightBrandColor, generic)
 
 		// Table view
-		self.tableBackgroundColor = UIColor.white
+		self.tableBackgroundColor = colors.resolveColor("Table.tableBackgroundColor", UIColor.white, generic)
+
 		if #available(iOS 13, *) {
-			self.tableGroupBackgroundColor = UIColor.groupTableViewBackground.resolvedColor(with: UITraitCollection(userInterfaceStyle: .light))
-			self.tableSeparatorColor = UIColor.separator
+			self.tableGroupBackgroundColor = colors.resolveColor("Table.tableGroupBackgroundColor", UIColor.groupTableViewBackground.resolvedColor(with: UITraitCollection(userInterfaceStyle: .light)), generic)
+			let color = colors.resolveColor("Table.tableSeparatorColor", UIColor.separator, generic)
+			self.tableSeparatorColor = color
+
 		} else {
-			self.tableGroupBackgroundColor = UIColor.groupTableViewBackground
-			self.tableSeparatorColor = UIColor.lightGray
+			self.tableGroupBackgroundColor = colors.resolveColor("Table.tableGroupBackgroundColor", UIColor.groupTableViewBackground, generic)
+			let color = colors.resolveColor("Table.tableSeparatorColor", UIColor.lightGray, generic)
+			self.tableSeparatorColor = color
 		}
 		self.tableSectionHeaderColor = UIColor.gray
 		self.tableSectionFooterColor = UIColor.gray
-		self.tableRowBorderColor = UIColor.black.withAlphaComponent(0.1)
 
-		self.tableRowColors = ThemeColorCollection(
+		let rowColor : UIColor? = UIColor.black.withAlphaComponent(0.1)
+		self.tableRowBorderColor = colors.resolveColor("Table.tableRowBorderColor", rowColor, generic)
+
+		self.tableRowColors = colors.resolveThemeColorCollection("Table.tableRowColors", ThemeColorCollection(
 			backgroundColor: tableBackgroundColor,
 			tintColor: nil,
 			labelColor: UIColor.black,
 			secondaryLabelColor: UIColor.gray,
 			symbolColor: darkColor,
 			filledColorPairCollection: ThemeColorPairCollection(fromPair: ThemeColorPair(foreground: UIColor.white, background: lightBrandColor))
-		)
+		), generic)
 
-		self.tableRowHighlightColors = ThemeColorCollection(
+		self.tableRowHighlightColors = colors.resolveThemeColorCollection("Table.tableRowHighlightColors", ThemeColorCollection(
 			backgroundColor: UIColor.white.darker(0.1),
 			tintColor: nil,
 			labelColor: UIColor.black,
 			secondaryLabelColor: UIColor.gray,
 			symbolColor: darkColor,
 			filledColorPairCollection: ThemeColorPairCollection(fromPair: ThemeColorPair(foreground: UIColor.white, background: lightBrandColor))
-		)
+		), generic)
+
+		self.favoriteEnabledColor = UIColor(hex: 0xFFCC00)
+		self.favoriteDisabledColor = UIColor(hex: 0x7C7C7C)
```

Comment:
This line is a duplicate (identical line below).

Reply:
Fixed